### PR TITLE
[DOCS] Clarify temporary directory location and retention

### LINKED ISSUE #11789: Document temp dir retention in how-to guide
I discovered by accident Pytest's feature to retain temporary directories by default after tests complete. I feel the how-to guide "How to use temporary directories and files in tests" (https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) could do with documenting this feature as this may be the first and only page a developer reads before using the `tmp_path` fixture.

Currently, the documentation simply refers to "a temporary directory unique to the test (...)". As an engineer, I read that as something that goes away automatically as soon as the fixture is being tore down.

This may be an issue when using large temporary files, for example, in automated pipelines. 

I don't mind what wording is used, but a reference to the API reference docs (e.g. https://docs.pytest.org/en/7.3.x/reference/reference.html#confval-tmp_path_retention_policy) might be extremely useful.

### PR DESCRIPTION
**Pull request by faph**
Fixes #11789 and #11790

### CODE REVIEW

File: doc/en/how-to/tmp_path.rst [hunk 1]
```diff
@@ -8,9 +8,13 @@ How to use temporary directories and files in tests
 The ``tmp_path`` fixture
 ------------------------
 
-You can use the ``tmp_path`` fixture which will
-provide a temporary directory unique to the current test,
-created in the `base temporary directory`_.
+You can use the ``tmp_path`` fixture which will provide a temporary directory
+unique to each test function.
+
+By default, ``pytest`` retains the temporary directory for the last 3 ``pytest``
```
Comment (nicoddemus) [inline doc/en/how-to/tmp_path.rst:14]: This looks good, I would just move this to the end of this session (before line `69`), as this seems more like a detail best left for after we explain how to use the fixture.

Comment (faph) [reply doc/en/how-to/tmp_path.rst]: Agreed and updated.

**Review by nicoddemus — APPROVED** [review]
Thanks!

**Comment by faph** [comment]
Awesome. Thanks @nicoddemus

**Pull request merged**